lms/templates/learner_dashboard/program_details_tab_view.underscore

```diff
@@ -47,7 +47,47 @@
                         </div>
                     </div>
                 <% } %>
-                <% if (is_learner_eligible_for_one_click_purchase && (typeof is_mobile_only === 'undefined' || is_mobile_only === false)) { %>
+                <% if (isSubscriptionEligible) { %>
+                <div class="d-flex flex-column align-items-start flex-sm-row  align-items-sm-center upgrade-subscription">
+                    <a href="<%- subscriptionUrl %>" class="btn-brand btn cta-primary upgrade-button">
+                    <% if (subscriptionState === 'inactive') { %>
+                        <div class="d-flex align-items-center">
+                            <div class="subscription-icon-restart">
+                                <% // xss-lint: disable=underscore-not-escaped %>
+                                <%= restartIcon %>
+                            </div>
+                            <span><%- gettext('Restart my subscription') %></span>
+                        </div>
+                    <% } else { %>
+                        <%- StringUtils.interpolate(gettext(
+                            subscriptionState === 'active'
+                            ? 'Manage my subscription'
+                            : 'Start {trialLength}-Day free trial'
+                        ),
+                        { trialLength } ) %>
+                    <% } %>
+                    </a>
+                    <span class="subscription-info-brief">
+                        <%- StringUtils.interpolate(
+                            gettext(
+                                hasActiveTrial
+                                ? 'Active trial ends {trialEndDate} at {trialEndTime}'
+                                : subscriptionState === 'active'
+                                ? 'Your next billing date is {nextPaymentDate}'
+                                : subscriptionState === 'inactive'
+                                ? 'Unlock verified access to all courses for {subscriptionPrice}/month. Cancel anytime.'
+                                : '{subscriptionPrice}/month subscription after trial ends. Cancel anytime.'
+                            ),
```

Comment:
nit: Another place that I find it a bit hard to understand what is going on because of the chained expressions. I would love to see this broken up if possible

Reply:
To add on, what is the condition line 79 is fulfilling, and how is it different from when `hasActiveTrial` is true?

Reply:
@justinhynes we can just read this as multiple if and else statements, I wouldn't want to break this up further,

@jsnwesson we have 4 possible states 
So firstly we check if the subscription is active and the user is using a free trial,
Then we check if the subscription is active (without free trial, paid subscription)
Then we check if the user has inactive subscription (this can be multiple cases,
User ended his subscription/subscription has paused because automatic payment from user failed etc)
the last state is the default initial state, in which the user has no past records of subscription to the program, so they can start a 7 day trial by clicking the cta